### ISSUE TITLE
Terraform re-apply causes cluster deletion and recreation due to `datapath_provider` parameter

### ISSUE THREAD

**Issue by NibiruHeisenberg** [terraform-google-modules/terraform-google-kubernetes-engine#1446]
### TL;DR

Even when the default value for the `datapath_provider` variable is `DATAPATH_PROVIDER_UNSPECIFIED`, the `beta-autopilot-public-cluster` module creates a GKE cluster with Dataplane V2 enabled (`ADVANCED_DATAPATH`). 

This causes the next Terraform plan/apply to think that an outside change was made to the GKE cluster and forces a destroy + recreate.

### Expected behavior

When no change is made to the Terraform config, nothing changes in the Terraform plan.

### Observed behavior

The Terraform plan output is saying that it will replace the cluster resource (delete + recreate) due to the `datapath_provider` variable

### Terraform Configuration

```hcl
locals {
  network_name           = "gke-heisenberg-public-network"
  subnet_name            = "gke-heisenberg-public-subnet"
  master_auth_subnetwork = "gke-heisenberg-public-master-subnet"
  pods_range_name        = "ip-range-pods-gke-heisenberg-public"
  svc_range_name         = "ip-range-svc-gke-heisenberg-public"
  subnet_names           = [for subnet_self_link in module.gcp-network.subnets_self_links : split("/", subnet_self_link)[length(split("/", subnet_self_link)) - 1]]
}


module "gcp-network" {
  source       = "terraform-google-modules/network/google"
  project_id   = var.project_id
  network_name = local.network_name

  subnets = [
    {
      subnet_name   = local.subnet_name
      subnet_ip     = "10.0.0.0/17"
      subnet_region = var.region
    },
    {
      subnet_name   = local.master_auth_subnetwork
      subnet_ip     = "10.60.0.0/17"
      subnet_region = var.region
    },
  ]

  secondary_ranges = {
    (local.subnet_name) = [
      {
        range_name    = local.pods_range_name
        ip_cidr_range = "192.168.0.0/18"
      },
      {
        range_name    = local.svc_range_name
        ip_cidr_range = "192.168.64.0/18"
      },
    ]
  }
}


module "gke" {
  source                          = "terraform-google-modules/kubernetes-engine/google//modules/beta-autopilot-public-cluster"
  project_id                      = var.project_id
  name                            = "gke-heisenberg-cluster"
  regional                        = true
  region                          = var.region
  network                         = module.gcp-network.network_name
  subnetwork                      = local.subnet_names[index(module.gcp-network.subnets_names, local.subnet_name)]
  ip_range_pods                   = local.pods_range_name
  ip_range_services               = local.svc_range_name
  release_channel                 = "RAPID"
  enable_vertical_pod_autoscaling = true
}
```

### Terraform Version

```sh
1.3.3
```

### Additional information

Region is us-east1.

**Comment by grsiepka** [comment]
Don't have any data to share ATM but wanted to report we are seeing the same exact issue using the **beta-autopilot-private-cluster** module as well.

**Comment by bharathkkb** [comment]
Thanks for the report @NibiruHeisenberg  @grsiepka. I suspect this is an issue with the upstream API changing the default. [Per docs](https://cloud.google.com/kubernetes-engine/docs/how-to/dataplane-v2), I see `GKE Dataplane V2 enabled in versions 1.22.7-gke.1500 and later and versions 1.23.4-gke.1500 and later`. I was able to repro this using just the resource 

```tf
resource "google_container_cluster" "primary" {
  name               = "my-gke-cluster"
  location           = "us-central1"
  project            = "test-datapath-gkecl"
  initial_node_count = 1
  datapath_provider  = "DATAPATH_PROVIDER_UNSPECIFIED"
  release_channel {
    channel = "RAPID"
  }
  enable_autopilot = true
  ip_allocation_policy {
  }
}
```
which resulted in the cluster getting created with ` "datapath_provider": "ADVANCED_DATAPATH"`. I would recommend opening an issue on the provider repo for confirmation. 

I suspect the delete recreate issue is due to a recent provider update https://github.com/GoogleCloudPlatform/magic-modules/pull/6657

**Comment by mastersingh24** [comment]
You can't set `datapath_provider` on an Autopilot cluster.  Hopefully the initial fix for this is simply removing it from this module.

**Comment by bharathkkb** [comment]
@mastersingh24 is the AP default `ADVANCED_DATAPATH`? That would explain what @grsiepka observed.

**Comment by mastersingh24** [comment]
> > You can't set datapath_provider on an Autopilot cluster.
> 
> Are you sure about that?
> 

Yep.  I'm sure. 

Just to clarify, setting `datapath_provider` on an Autopilot cluster actually has no affect on the cluster itself.  No matter what you pass in, it will be ignored.  All Autopilot clusters created for almost the past year use Dataplane V2.  Clusters created prior to that did not, but they are all currently being transparently migrated during upgrades.



> I've built a handful of AutoPilot clusters since this initial report, setting "datapath_provider" to "ADVANCED_DATAPATH" does not cause the cluster to get re-created on following plans/applies.
> 

The problem occurs if you
a) don't have `datapath_provider` in your config
b) have `datapath_provider` in your config but not set to "ADVANCED_DATAPATH"

For a), [the module currently defaults to "DATAPATH_PROVIDER_UNSPECIFIED"](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/variables.tf#L108), so when comparing against clusters created in the last year, it does not match and since its immutable it is marked to be recreated

For b), if you set it to "DATAPATH_PROVIDER_UNSPECIFIED" in your config, it's actually ignored by the GKE API when creating the Autopilot cluster (which sets it to "ADVANCED_DATAPATH").  And so once again, you have a mismatch if you try to perform an update on a mutable field.

**Comment by grsiepka** [comment]
Ah that makes more sense now, thanks for the clarification.
